### ISSUE TITLE
Add more clarity/info on the -fsycl-device-code-split=<value> flag

### ISSUE THREAD

**Issue by Srividya2212** [intel/llvm#1489]
`clang++ --help` did not list `-fsycl-device-code-split=<value>`.
The description provided in the [user-manual](https://github.com/intel/llvm/blob/7a2e75e57d224436468f263c9e3385b1ea6b8c25/sycl/doc/UsersManual.md) is not very helpful. For example : what is a `device code module`? What does `per-source basis` mean? A much elaborate description for this flag and what each of the 3 values mean would be helpful to better understand the usage of the `-fsycl-device-code-split` flag.

I am assigning this issue based on the contributors list I saw on the user manual doc.

**Comment by alexbatashev** [comment]
I definitely see the documentation bits in the code:  https://github.com/intel/llvm/blob/803f6cc552e8b557d05958a1aa06e44bad0294ae/clang/include/clang/Driver/Options.td#L1857-L1862

@AGindinson @mdtoguchi do you know why they are not shown?

**Comment by AGindinson** [comment]
> Do you know why they are not shown?
Aren't they?
```bash
clang++ --help | grep device-code-split -A 1
```
prints this information precisely.

**Comment by Fznamznon** [comment]
> > Do you know why they are not shown?
> 
> Aren't they?
> 
> ```shell
> clang++ --help | grep device-code-split -A 1
> ```
> 
> prints this information precisely.

Yep, I can see it too.

**Comment by Srividya2212** [comment]
This is my `clang++` version
```
sundaram@scsel-cfl-12:/iusers/sundaram/8205$ clang++ --version
clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```
And my result for `clang++ --help | grep device-code-split -A 1` is empty:
![clang-help](https://user-images.githubusercontent.com/8020274/78702196-11bd6b80-78bd-11ea-8201-5767aa5f563e.JPG)

Am I missing something here?
Also, does clang++ help print the exact same description provided in the User Manual? If yes, then it would be helpful to update the description and address some questions ( please see my original issue description).

**Comment by AGindinson** [comment]
@Srrividya2212, it appears that you're using the llvm.org release version of the compiler. Our DPC++ implementation is not part of the llvm.org codebase _yet_, so you would need to build the compiler manually as per our [Get Started Guide](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md) and make sure that the resulting binaries are called instead of the vanilla Clang.

**Comment by alexbatashev** [comment]
@Srividya2212 please reopen if you still have problems.

**Comment by Fznamznon** [comment]
> This is my `clang++` version
> 
> ```
> sundaram@scsel-cfl-12:/iusers/sundaram/8205$ clang++ --version
> clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
> Target: x86_64-pc-linux-gnu
> Thread model: posix
> InstalledDir: /usr/bin
> ```
> 
> And my result for `clang++ --help | grep device-code-split -A 1` is empty:
> ![clang-help](https://user-images.githubusercontent.com/8020274/78702196-11bd6b80-78bd-11ea-8201-5767aa5f563e.JPG)
> 
> Am I missing something here?

You are running some old version of clang installed into system. Probably you forgot to set `PATH` env variable to local build or dpc++ installation.

> Also, does clang++ help print the exact same description provided in the User Manual? If yes, then it would be helpful to update the description and address some questions ( please see my original issue description).

It doesn't. `clang++` prints reduced description comparing with User Manual.
Actually for me User Manual seem describing all the option and "what is per_source basis" pretty well, doesn't it?

>- per_source - a separate device code module is created for each source
(translation unit). Each device code module will contain a bunch of kernels
grouped on per-source basis and all their dependencies, i.e. all used
variables and called functions, including the `SYCL_EXTERNAL` macro-marked
functions from other translation units.

The `device code module` term is not explained, that is right. But all that it means is SPIR-V or LLVM IR module containing device code. I don't think that it deserves extended explanation.

**Comment by Srividya2212** [comment]
Thanks for the clarification. The main reason to open this issue was to get clarification on the few terms used in the description of the flag. It might be obvious for folks working on compiler but few of us who are new to compiler development couldn't make complete sense of those terms. Your explanation for "device code module" makes sense.
